Gems/Atom/Utils/Code/Source/ImGuiGpuProfiler.cpp

```diff
@@ -1366,18 +1402,131 @@ namespace AZ
             {
                 if (ImGui::Button("Capture"))
                 {
-                    // Collect and save new GPU memory usage data
-                    const auto* memoryStatistics = rhiSystem->GetMemoryStatistics();
-                    if (memoryStatistics) 
+                    m_showingLoadedCapture = false;
+                    PerformCapture();
+                }
+
+                ImGui::SameLine();
+
+                if (ImGui::Button("Save to CSV"))
+                {
+                    if (m_savedPools.empty())
+                    {
+                        PerformCapture();
+                    }
+
+                    SaveToCSV();
+                }
+                ImGui::SameLine();
+                constexpr static const char* LoadMemoryCaptureTitle = "Select or input memory capture csv file";
+                if (ImGui::Button("Load from CSV"))
+                {
+                    m_captureInput[0] = '\0';
+                    m_captureSelection = 0;
+                    ImGui::OpenPopup(LoadMemoryCaptureTitle);
+                }
+
+                // Always center this window when appearing
+                ImVec2 center = ImGui::GetMainViewport()->GetCenter();
+                ImGui::SetNextWindowPos(center, ImGuiCond_Appearing, ImVec2(0.5f, 0.5f));
+
+                if (ImGui::BeginPopupModal(LoadMemoryCaptureTitle, nullptr, ImGuiWindowFlags_AlwaysAutoResize))
+                {
+                    struct CaptureInfo
+                    {
+                        AZStd::string path;
+                        std::filesystem::file_time_type time;
+                    };
+                    AZStd::vector<CaptureInfo> captures;
+
+                    // Enumerate files in the capture folder
+                    for (const std::filesystem::directory_entry& entry : std::filesystem::directory_iterator{ m_memoryCapturePath.c_str() })
+                    {
+                        if (entry.is_regular_file())
+                        {
+                            if (entry.path().extension() == ".csv")
+                            {
+                                captures.push_back(CaptureInfo{ entry.path().string().c_str(), entry.last_write_time() });
+                            }
+                        }
+                    }
```

Comment:
This could be replaced with AZ equivalents as such:
```c++
AZStd::vector<AZ::IO::Path> captures;

auto* base = AZ::IO::FileIOBase::GetInstance();
base->FindFiles(m_memoryCapturePath.c_str(), "*.csv",
    [&captures](const char* path) -> bool
    {
        auto foundPath = AZ::IO::Path(path);
        captures.push_back(foundPath);
        return true; // continue iteration
    });
```